pandas/core/groupby/generic.py

```diff
@@ -1754,10 +1754,22 @@ def value_counts(
                     level=index_level, sort_remaining=False
                 )
 
-            if not self.as_index:
+            if self.as_index:
+                return result.__finalize__(self.obj, method="value_counts")
```

Comment:
do `result_frame = result` and then do the `__finalize__` after the if/then (e.g. share it between these)

Reply:
Done